Tests do not pass with current version of Pygments

Using Pygments 2.2.0 (the current version at this time), some code block tests fail (apparently because of minor changes to Pygments ouput). They work fine using, e.g., Pygments 2.0.2.

Sample diff:
```
    --- ./tm-cases/fenced_code_blocks_leading_lang_space.html
    +++ markdown2 ./tm-cases/fenced_code_blocks_leading_lang_space.text
    @@ -1,3 +1,3 @@
    -<div class="codehilite"><pre><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    -    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    +    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
     </code></pre></div>
```

7 tests seem to fail in total, all of them in this category.

Pinning version in CI is not an ideal solution ;-) Actually trying to refresh our openSUSE package failed due to errors in tests too.

This is the python(2,3) version used in our build system.
```
libpython2_7-1_0-2.7.14-2.2.x86_64
libpython3_6m1_0-3.6.4-1.3.x86_64
python-2.7.14-2.2.x86_64
python-base-2.7.14-2.2.x86_64
python-devel-2.7.14-2.2.x86_64
python-rpm-macros-2017.12.22.d9968ab-56.2.noarch
python-xml-2.7.14-2.2.x86_64
python2-Pygments-2.2.0-53.7.noarch
python2-appdirs-1.4.3-8.10.noarch
python2-packaging-16.8-20.15.noarch
python2-pyparsing-2.2.0-1.15.noarch
python2-setuptools-38.4.0-112.2.noarch
python2-six-1.11.0-66.6.noarch
python3-3.6.4-1.2.x86_64
python3-Pygments-2.2.0-53.7.noarch
python3-appdirs-1.4.3-8.10.noarch
python3-base-3.6.4-1.3.x86_64
python3-devel-3.6.4-1.3.x86_64
python3-packaging-16.8-20.15.noarch
python3-pyparsing-2.2.0-1.15.noarch
python3-setuptools-38.4.0-112.2.noarch
python3-six-1.11.0-66.6.noarch
```

And here the tests failing with markdown2 2.3.5
**Summary**
```
+ echo python2
+ export PYTHONPATH=/home/abuild/rpmbuild/BUILDROOT/python-markdown2-2.3.5-0.x86_64/usr/lib/python2.7/site-packages
+ PYTHONPATH=/home/abuild/rpmbuild/BUILDROOT/python-markdown2-2.3.5-0.x86_64/usr/lib/python2.7/site-packages
+ python2 test.py -- -knownfailure
markdown2/tm/ampersands [htmlentities] ... ok
markdown2/tm/auto_link ... ok
markdown2/tm/auto_link_email_with_underscore [issue26] ... ok
markdown2/tm/auto_link_safe_mode [issue7, safe_mode] ... ok
markdown2/tm/basic_safe_mode [safe_mode] ... ok
markdown2/tm/basic_safe_mode_escape [safe_mode] ... ok
markdown2/tm/blockquote ... ok
markdown2/tm/blockquote_with_pre ... ok
markdown2/tm/break_on_newline ... ok
markdown2/tm/code_block_with_tabs [fromphpmarkdown] ... ok
markdown2/tm/code_blocks_leading_line ... ok
markdown2/tm/code_safe_emphasis [code_friendly] ... ok
markdown2/tm/codeblock ... ok
markdown2/tm/codespans ... ok
markdown2/tm/codespans_safe_mode [issue9, safe_mode] ... ok
markdown2/tm/cuddled_list_indented [smedberg, issue36, extra, cuddle-lists] ... ok
markdown2/tm/cuddled_para_and_list [smedberg, issue33, extra, cuddle-lists] ... ok
markdown2/tm/cuddled_with_para ... ok
markdown2/tm/demote_headers ... ok
markdown2/tm/emacs_head_vars [emacs] ... ok
markdown2/tm/emacs_tail_vars [emacs] ... ok
markdown2/tm/emphasis ... ok
markdown2/tm/escapes [issue15] ... ok
markdown2/tm/fenced_code_blocks_issue113 [extra, fenced-code-blocks, issue113] ... ok
markdown2/tm/fenced_code_blocks_issue127 [extra, fenced-code-blocks, issue127] ... ok
markdown2/tm/fenced_code_blocks_issue135 [extra, fenced-code-blocks, issue135] ... ok
markdown2/tm/fenced_code_blocks_issue161 [extra, fenced-code-blocks] ... ok
markdown2/tm/fenced_code_blocks_issue86 [extra, fenced-code-blocks, issue86, issue84, issue87] ... ok
markdown2/tm/fenced_code_blocks_leading_lang_space [extra, fenced-code-blocks] ... FAIL
markdown2/tm/fenced_code_blocks_leading_line [extra, fenced-code-blocks] ... ok
markdown2/tm/fenced_code_blocks_leading_paragraph [extra, fenced-code-blocks] ... ok
markdown2/tm/fenced_code_blocks_safe_highlight [extra, fenced-code-blocks, pygments] ... FAIL
markdown2/tm/fenced_code_blocks_simple [extra, fenced-code-blocks] ... ok
markdown2/tm/fenced_code_blocks_syntax_highlighting [extra, fenced-code-blocks, pygments] ... FAIL
markdown2/tm/fenced_code_blocks_syntax_indentation [extra, fenced-code-blocks, pygments, indentation] ... FAIL
markdown2/tm/footnotes ... ok
markdown2/tm/footnotes_custom ... ok
markdown2/tm/footnotes_letters [footnotes] ... ok
markdown2/tm/footnotes_markup [footnotes] ... ok
markdown2/tm/footnotes_safe_mode_escape [safe_mode, footnotes] ... ok
markdown2/tm/footnotes_underscores [issue27] ... ok
markdown2/tm/header ... ok
markdown2/tm/header_atx_no_preceeding_space ... ok
markdown2/tm/header_ids_1 [extra, header-ids] ... ok
markdown2/tm/header_ids_2 [extra, header-ids] ... ok
markdown2/tm/header_ids_3 [extra, header-ids] ... ok
markdown2/tm/header_ids_4 [extra, header-ids, unicode] ... ok
markdown2/tm/header_ids_5 [extra, header-ids] ... ok
markdown2/tm/headers_tag_friendly [extra, tag-friendly] ... ok
markdown2/tm/highlightjs_lang [extra, fenced-code-blocks, highlightjs-lang] ... ok
markdown2/tm/hr ... ok
markdown2/tm/hr_length ... ok
markdown2/tm/hr_spaces ... ok
markdown2/tm/html5_block_tags [issue57] ... ok
markdown2/tm/img_in_link ... ok
markdown2/tm/incorrect_list_parse [issue165] ... ok
markdown2/tm/inline_links ... ok
markdown2/tm/issue21_gt_escaping [issue21, escape] ... ok
markdown2/tm/issue2_safe_mode_borks_markup [safe_mode] ... ok
markdown2/tm/issue3_bad_code_color_hack [extra, code-color, unicode, pygments, issue3] ... FAIL
markdown2/tm/issue54_escape_link_title [issue54] ... ok
markdown2/tm/link_defn_alt_title_delims ... ok
markdown2/tm/link_defn_spaces_in_url [issue18] ... ok
markdown2/tm/link_nofollow [extras, issue74, issue104] ... ok
markdown2/tm/link_patterns ... ok
markdown2/tm/link_patterns_double_hit [link_patterns] ... ok
markdown2/tm/link_patterns_edge_cases [link_patterns] ... ok
markdown2/tm/link_with_blank [extras, issue213] ... ok
markdown2/tm/lists ... ok
markdown2/tm/lists2 [issue16] ... ok
markdown2/tm/long_link [issue24] ... ok
markdown2/tm/markdown_in_html [markdown-in-html, toc, extra] ... ok
markdown2/tm/metadata [extra, metadata, issue78] ... ok
markdown2/tm/metadata2 [extra, metadata, issue78] ... ok
markdown2/tm/mismatched_footnotes ... ok
markdown2/tm/missing_link_defn ... ok
markdown2/tm/nested_list ... ok
markdown2/tm/nested_list_safe_mode [issue9, safe_mode] ... ok
markdown2/tm/not_quite_a_list ... ok
markdown2/tm/numbering ... ok
markdown2/tm/parens_in_url_4 [fromphpmarkdown] ... ok
markdown2/tm/pi_and_xinclude [xml, pi, xinclude] ... ok
markdown2/tm/pyshell ... ok
markdown2/tm/raw_html ... ok
markdown2/tm/ref_links ... ok
markdown2/tm/smarty_pants [smarty-pants, issue42] ... ok
markdown2/tm/smarty_pants_image_links [smarty-pants, issue76] ... ok
markdown2/tm/spoiler ... ok
markdown2/tm/strike [extras, strike] ... ok
markdown2/tm/sublist-ordered-para [issue67] ... ok
markdown2/tm/sublist-para [questionable] ... ok
markdown2/tm/syntax_color [extra, code-color, pygments] ... FAIL
markdown2/tm/syntax_color_opts [extra, code-color, pygments] ... FAIL
markdown2/tm/tables [extra, tables] ... ok
markdown2/tm/task_list [extra, task_list, issue216] ... ok
markdown2/tm/toc_1 [toc, extra] ... ok
markdown2/tm/toc_2 [toc, extra] ... ok
markdown2/tm/toc_3 [toc, extra] ... ok
markdown2/tm/toc_4 [toc, extra] ... ok
markdown2/tm/tricky_anchors ... ok
markdown2/tm/two_comments ... ok
markdown2/tm/underline_in_autolink ... ok
markdown2/tm/wiki_tables [extra, wiki-tables, issue66] ... ok
markdown2/tm/xss_quotes [issue30] ... ok
markdown2/direct/code_in_strong [code, strong] ... ok
markdown2/direct/pre ... ok
markdown2/direct/russian [unicode, issue3] ... ok
markdown2/direct/slow_hr [perf] ... ok
markdown2/direct/starter_pre [pre, recipes] ... ok
markdown2/doctests/api ... ok
markdown2/doctests/internal ... ok
```
**Details**
```
markdown2/doctests/api ... ok
markdown2/doctests/internal ... ok

======================================================================
FAIL: markdown2/tm/fenced_code_blocks_leading_lang_space [extra, fenced-code-blocks]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    ```.python\n
    if.True:\n
    ....print."hi"\n
    ```\n
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><span></span><code><span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s2">&quot;hi&quot;</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><code><span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s">&quot;hi&quot;</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/fenced_code_blocks_leading_lang_space.html
    +++ markdown2 ./tm-cases/fenced_code_blocks_leading_lang_space.text
    @@ -1,3 +1,3 @@
    -<div class="codehilite"><pre><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    -    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    +    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
     </code></pre></div>


======================================================================
FAIL: markdown2/tm/fenced_code_blocks_safe_highlight [extra, fenced-code-blocks, pygments]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    ```python\n
    if.True:\n
    ....print."hi"\n
    ```\n
    \n
    That's.using.the.*fenced-code-blocks*.extra.with.Python\n
    syntax.coloring,.if.`pygments`.is.installed..See\n
    <http://github.github.com/github-flavored-markdown/>.\n
    \n
    ```ruby\n
    def.foo\n
    ....puts."hi"\n
    end\n
    ```\n
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><span></span><code><span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s2">&quot;hi&quot;</span>\n
    </code></pre></div>\n
    \n
    <p>That's.using.the.<em>fenced-code-blocks</em>.extra.with.Python\n
    syntax.coloring,.if.<code>pygments</code>.is.installed..See\n
    <a.href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>\n
    \n
    <div.class="codehilite"><pre><span></span><code><span.class="k">def</span>.<span.class="nf">foo</span>\n
    ....<span.class="nb">puts</span>.<span.class="s2">&quot;hi&quot;</span>\n
    <span.class="k">end</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><code><span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s">&quot;hi&quot;</span>\n
    </code></pre></div>\n
    \n
    <p>That's.using.the.<em>fenced-code-blocks</em>.extra.with.Python\n
    syntax.coloring,.if.<code>pygments</code>.is.installed..See\n
    <a.href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>\n
    \n
    <div.class="codehilite"><pre><code><span.class="k">def</span>.<span.class="nf">foo</span>\n
    ....<span.class="nb">puts</span>.<span.class="s2">&quot;hi&quot;</span>\n
    <span.class="k">end</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/fenced_code_blocks_safe_highlight.html
    +++ markdown2 ./tm-cases/fenced_code_blocks_safe_highlight.text
    @@ -1,12 +1,12 @@
    -<div class="codehilite"><pre><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    -    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    +    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
     </code></pre></div>
     
     <p>That's using the <em>fenced-code-blocks</em> extra with Python
     syntax coloring, if <code>pygments</code> is installed. See
     <a href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>
     
    -<div class="codehilite"><pre><code><span class="k">def</span> <span class="nf">foo</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">def</span> <span class="nf">foo</span>
         <span class="nb">puts</span> <span class="s2">&quot;hi&quot;</span>
     <span class="k">end</span>
     </code></pre></div>


======================================================================
FAIL: markdown2/tm/fenced_code_blocks_syntax_highlighting [extra, fenced-code-blocks, pygments]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    ```python\n
    if.True:\n
    ....print."hi"\n
    ```\n
    \n
    That's.using.the.*fenced-code-blocks*.extra.with.Python\n
    syntax.coloring,.if.`pygments`.is.installed..See\n
    <http://github.github.com/github-flavored-markdown/>.\n
    \n
    ```ruby\n
    def.foo\n
    ....puts."hi"\n
    end\n
    ```\n
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><span></span><code><span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s2">&quot;hi&quot;</span>\n
    </code></pre></div>\n
    \n
    <p>That's.using.the.<em>fenced-code-blocks</em>.extra.with.Python\n
    syntax.coloring,.if.<code>pygments</code>.is.installed..See\n
    <a.href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>\n
    \n
    <div.class="codehilite"><pre><span></span><code><span.class="k">def</span>.<span.class="nf">foo</span>\n
    ....<span.class="nb">puts</span>.<span.class="s2">&quot;hi&quot;</span>\n
    <span.class="k">end</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><code><span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s">&quot;hi&quot;</span>\n
    </code></pre></div>\n
    \n
    <p>That's.using.the.<em>fenced-code-blocks</em>.extra.with.Python\n
    syntax.coloring,.if.<code>pygments</code>.is.installed..See\n
    <a.href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>\n
    \n
    <div.class="codehilite"><pre><code><span.class="k">def</span>.<span.class="nf">foo</span>\n
    ....<span.class="nb">puts</span>.<span.class="s2">&quot;hi&quot;</span>\n
    <span.class="k">end</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/fenced_code_blocks_syntax_highlighting.html
    +++ markdown2 ./tm-cases/fenced_code_blocks_syntax_highlighting.text
    @@ -1,12 +1,12 @@
    -<div class="codehilite"><pre><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    -    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    +    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
     </code></pre></div>
     
     <p>That's using the <em>fenced-code-blocks</em> extra with Python
     syntax coloring, if <code>pygments</code> is installed. See
     <a href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>
     
    -<div class="codehilite"><pre><code><span class="k">def</span> <span class="nf">foo</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">def</span> <span class="nf">foo</span>
         <span class="nb">puts</span> <span class="s2">&quot;hi&quot;</span>
     <span class="k">end</span>
     </code></pre></div>


======================================================================
FAIL: markdown2/tm/fenced_code_blocks_syntax_indentation [extra, fenced-code-blocks, pygments, indentation]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    ```python\n
    def.foo():\n
    ....print."foo"\n
    \n
    ....print."bar"\n
    ```\n
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><span></span><code><span.class="k">def</span>.<span.class="nf">foo</span><span.class="p">():</span>\n
    ....<span.class="k">print</span>.<span.class="s2">&quot;foo&quot;</span>\n
    \n
    ....<span.class="k">print</span>.<span.class="s2">&quot;bar&quot;</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><code><span.class="k">def</span>.<span.class="nf">foo</span><span.class="p">():</span>\n
    ....<span.class="k">print</span>.<span.class="s">&quot;foo&quot;</span>\n
    \n
    ....<span.class="k">print</span>.<span.class="s">&quot;bar&quot;</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/fenced_code_blocks_syntax_indentation.html
    +++ markdown2 ./tm-cases/fenced_code_blocks_syntax_indentation.text
    @@ -1,5 +1,5 @@
    -<div class="codehilite"><pre><code><span class="k">def</span> <span class="nf">foo</span><span class="p">():</span>
    -    <span class="k">print</span> <span class="s">&quot;foo&quot;</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">def</span> <span class="nf">foo</span><span class="p">():</span>
    +    <span class="k">print</span> <span class="s2">&quot;foo&quot;</span>
     
    -    <span class="k">print</span> <span class="s">&quot;bar&quot;</span>
    +    <span class="k">print</span> <span class="s2">&quot;bar&quot;</span>
     </code></pre></div>


======================================================================
FAIL: markdown2/tm/issue3_bad_code_color_hack [extra, code-color, unicode, pygments, issue3]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    <!--.-*-.coding:.utf-8.-*-.-->\n
    \n
    ##.\u0437\u0430\u0433\u043e\u043b\u043e\u0432\u043e\u043a\n
    \n
    Example.from.<http://code.google.com/p/python-markdown2/issues/detail?id=3#c8>.\n
    \n
    Some.python.code:\n
    \n
    ....:::python\n
    ....#.\u043a\u043e\u043c\u043c\u0435\u043d\u0442\u0430\u0440\u0438\u0439\n
    ....if.True:\n
    ........print."hi"
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <!--.-*-.coding:.utf-8.-*-.-->\n
    \n
    <h2>\u0437\u0430\u0433\u043e\u043b\u043e\u0432\u043e\u043a</h2>\n
    \n
    <p>Example.from.<a.href="http://code.google.com/p/python-markdown2/issues/detail?id=3#c8">http://code.google.com/p/python-markdown2/issues/detail?id=3#c8</a>.</p>\n
    \n
    <p>Some.python.code:</p>\n
    \n
    <div.class="codehilite"><pre><span></span><code><span.class="c1">#.\u043a\u043e\u043c\u043c\u0435\u043d\u0442\u0430\u0440\u0438\u0439</span>\n
    <span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s2">&quot;hi&quot;</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <!--.-*-.coding:.utf-8.-*-.-->\n
    \n
    <h2>\u0437\u0430\u0433\u043e\u043b\u043e\u0432\u043e\u043a</h2>\n
    \n
    <p>Example.from.<a.href="http://code.google.com/p/python-markdown2/issues/detail?id=3#c8">http://code.google.com/p/python-markdown2/issues/detail?id=3#c8</a>.</p>\n
    \n
    <p>Some.python.code:</p>\n
    \n
    <div.class="codehilite"><pre><code><span.class="c">#.\u043a\u043e\u043c\u043c\u0435\u043d\u0442\u0430\u0440\u0438\u0439</span>\n
    <span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s">&quot;hi&quot;</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/issue3_bad_code_color_hack.html
    +++ markdown2 ./tm-cases/issue3_bad_code_color_hack.text
    @@ -6,7 +6,7 @@
     
     <p>Some python code:</p>
     
    -<div class="codehilite"><pre><code><span class="c"># \u043a\u043e\u043c\u043c\u0435\u043d\u0442\u0430\u0440\u0438\u0439</span>
    +<div class="codehilite"><pre><span></span><code><span class="c1"># \u043a\u043e\u043c\u043c\u0435\u043d\u0442\u0430\u0440\u0438\u0439</span>
     <span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    -    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
    +    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
     </code></pre></div>


======================================================================
FAIL: markdown2/tm/syntax_color [extra, code-color, pygments]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    Here.is.some.sample.code:\n
    \n
    ....:::python\n
    ....import.sys\n
    ....def.main(argv=sys.argv):\n
    ........logging.basicConfig()\n
    ........log.info('hi')\n
    \n
    and:\n
    \n
    ....:::ruby\n
    ....use.'zlib'\n
    ....sub.main(argv)\n
    ........puts.'hi'\n
    ....end\n
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <p>Here.is.some.sample.code:</p>\n
    \n
    <div.class="codehilite"><pre><span></span><code><span.class="kn">import</span>.<span.class="nn">sys</span>\n
    <span.class="k">def</span>.<span.class="nf">main</span><span.class="p">(</span><span.class="n">argv</span><span.class="o">=</span><span.class="n">sys</span><span.class="o">.</span><span.class="n">argv</span><span.class="p">):</span>\n
    ....<span.class="n">logging</span><span.class="o">.</span><span.class="n">basicConfig</span><span.class="p">()</span>\n
    ....<span.class="n">log</span><span.class="o">.</span><span.class="n">info</span><span.class="p">(</span><span.class="s1">&#39;hi&#39;</span><span.class="p">)</span>\n
    </code></pre></div>\n
    \n
    <p>and:</p>\n
    \n
    <div.class="codehilite"><pre><span></span><code><span.class="n">use</span>.<span.class="s1">&#39;zlib&#39;</span>\n
    <span.class="nb">sub</span>.<span.class="n">main</span><span.class="p">(</span><span.class="n">argv</span><span.class="p">)</span>\n
    ....<span.class="nb">puts</span>.<span.class="s1">&#39;hi&#39;</span>\n
    <span.class="k">end</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <p>Here.is.some.sample.code:</p>\n
    \n
    <div.class="codehilite"><pre><code><span.class="kn">import</span>.<span.class="nn">sys</span>\n
    <span.class="k">def</span>.<span.class="nf">main</span><span.class="p">(</span><span.class="n">argv</span><span.class="o">=</span><span.class="n">sys</span><span.class="o">.</span><span.class="n">argv</span><span.class="p">):</span>\n
    ....<span.class="n">logging</span><span.class="o">.</span><span.class="n">basicConfig</span><span.class="p">()</span>\n
    ....<span.class="n">log</span><span.class="o">.</span><span.class="n">info</span><span.class="p">(</span><span.class="s">&#39;hi&#39;</span><span.class="p">)</span>\n
    </code></pre></div>\n
    \n
    <p>and:</p>\n
    \n
    <div.class="codehilite"><pre><code><span.class="n">use</span>.<span.class="s1">&#39;zlib&#39;</span>\n
    <span.class="nb">sub</span>.<span.class="n">main</span><span.class="p">(</span><span.class="n">argv</span><span.class="p">)</span>\n
    ....<span.class="nb">puts</span>.<span.class="s1">&#39;hi&#39;</span>\n
    <span.class="k">end</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/syntax_color.html
    +++ markdown2 ./tm-cases/syntax_color.text
    @@ -1,14 +1,14 @@
     <p>Here is some sample code:</p>
     
    -<div class="codehilite"><pre><code><span class="kn">import</span> <span class="nn">sys</span>
    +<div class="codehilite"><pre><span></span><code><span class="kn">import</span> <span class="nn">sys</span>
     <span class="k">def</span> <span class="nf">main</span><span class="p">(</span><span class="n">argv</span><span class="o">=</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">):</span>
         <span class="n">logging</span><span class="o">.</span><span class="n">basicConfig</span><span class="p">()</span>
    -    <span class="n">log</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s">&#39;hi&#39;</span><span class="p">)</span>
    +    <span class="n">log</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s1">&#39;hi&#39;</span><span class="p">)</span>
     </code></pre></div>
     
     <p>and:</p>
     
    -<div class="codehilite"><pre><code><span class="n">use</span> <span class="s1">&#39;zlib&#39;</span>
    +<div class="codehilite"><pre><span></span><code><span class="n">use</span> <span class="s1">&#39;zlib&#39;</span>
     <span class="nb">sub</span> <span class="n">main</span><span class="p">(</span><span class="n">argv</span><span class="p">)</span>
         <span class="nb">puts</span> <span class="s1">&#39;hi&#39;</span>
     <span class="k">end</span>


======================================================================
FAIL: markdown2/tm/syntax_color_opts [extra, code-color, pygments]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    Here.is.some.sample.code:\n
    \n
    ....:::python\n
    ....import.sys\n
    ....def.main(argv=sys.argv):\n
    ........logging.basicConfig()\n
    ........log.info('hi')\n
    \n
    and:\n
    \n
    ....:::ruby\n
    ....use.'zlib'\n
    ....sub.main(argv)\n
    ........puts.'hi'\n
    ....end\n
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <p>Here.is.some.sample.code:</p>\n
    \n
    <div.class="codehilite".style="background:.#f8f8f8"><pre.style="line-height:.125%"><span></span><code><span.style="color:.#008000;.font-weight:.bold">import</span>.<span.style="color:.#0000FF;.font-weight:.bold">sys</span>\n
    <span.style="color:.#008000;.font-weight:.bold">def</span>.<span.style="color:.#0000FF">main</span>(argv<span.style="color:.#666666">=</span>sys<span.style="color:.#666666">.</span>argv):\n
    ....logging<span.style="color:.#666666">.</span>basicConfig()\n
    ....log<span.style="color:.#666666">.</span>info(<span.style="color:.#BA2121">&#39;hi&#39;</span>)\n
    </code></pre></div>\n
    \n
    <p>and:</p>\n
    \n
    <div.class="codehilite".style="background:.#f8f8f8"><pre.style="line-height:.125%"><span></span><code>use.<span.style="color:.#BA2121">&#39;zlib&#39;</span>\n
    <span.style="color:.#008000">sub</span>.main(argv)\n
    ....<span.style="color:.#008000">puts</span>.<span.style="color:.#BA2121">&#39;hi&#39;</span>\n
    <span.style="color:.#008000;.font-weight:.bold">end</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <p>Here.is.some.sample.code:</p>\n
    \n
    <div.class="codehilite".style="background:.#f8f8f8"><pre.style="line-height:.125%"><code><span.style="color:.#008000;.font-weight:.bold">import</span>.<span.style="color:.#0000FF;.font-weight:.bold">sys</span>\n
    <span.style="color:.#008000;.font-weight:.bold">def</span>.<span.style="color:.#0000FF">main</span>(argv<span.style="color:.#666666">=</span>sys<span.style="color:.#666666">.</span>argv):\n
    ....logging<span.style="color:.#666666">.</span>basicConfig()\n
    ....log<span.style="color:.#666666">.</span>info(<span.style="color:.#BA2121">&#39;hi&#39;</span>)\n
    </code></pre></div>\n
    \n
    <p>and:</p>\n
    \n
    <div.class="codehilite".style="background:.#f8f8f8"><pre.style="line-height:.125%"><code>use.<span.style="color:.#BA2121">&#39;zlib&#39;</span>\n
    <span.style="color:.#008000">sub</span>.main(argv)\n
    ....<span.style="color:.#008000">puts</span>.<span.style="color:.#BA2121">&#39;hi&#39;</span>\n
    <span.style="color:.#008000;.font-weight:.bold">end</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/syntax_color_opts.html
    +++ markdown2 ./tm-cases/syntax_color_opts.text
    @@ -1,6 +1,6 @@
     <p>Here is some sample code:</p>

    -<div class="codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><code><span style="color: #008000; font-weight: bold">import</span> <span style="color: #0000FF; font-weight: bold">sys</span>
    +<div class="codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><span></span><code><span style="color: #008000; font-weight: bold">import</span> <span style="color: #0000FF; font-weight: bold">sys</span>
     <span style="color: #008000; font-weight: bold">def</span> <span style="color: #0000FF">main</span>(argv<span style="color: #666666">=</span>sys<span style="color: #666666">.</span>argv):
         logging<span style="color: #666666">.</span>basicConfig()
         log<span style="color: #666666">.</span>info(<span style="color: #BA2121">&#39;hi&#39;</span>)
    @@ -8,7 +8,7 @@

     <p>and:</p>

    -<div class="codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><code>use <span style="color: #BA2121">&#39;zlib&#39;</span>
    +<div class="codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><span></span><code>use <span style="color: #BA2121">&#39;zlib&#39;</span>
     <span style="color: #008000">sub</span> main(argv)
         <span style="color: #008000">puts</span> <span style="color: #BA2121">&#39;hi&#39;</span>
     <span style="color: #008000; font-weight: bold">end</span>


----------------------------------------------------------------------
Ran 111 tests in 0.157s

FAILED (7 failures)
error: Bad exit status from /var/tmp/rpm-tmp.vaPyri (%check)
```

Pinning version in CI is not an ideal solution ;-) Actually trying to refresh our openSUSE package failed due to errors in tests too.

This is the python(2,3) version used in our build system.
```
libpython2_7-1_0-2.7.14-2.2.x86_64
libpython3_6m1_0-3.6.4-1.3.x86_64
python-2.7.14-2.2.x86_64
python-base-2.7.14-2.2.x86_64
python-devel-2.7.14-2.2.x86_64
python-rpm-macros-2017.12.22.d9968ab-56.2.noarch
python-xml-2.7.14-2.2.x86_64
python2-Pygments-2.2.0-53.7.noarch
python2-appdirs-1.4.3-8.10.noarch
python2-packaging-16.8-20.15.noarch
python2-pyparsing-2.2.0-1.15.noarch
python2-setuptools-38.4.0-112.2.noarch
python2-six-1.11.0-66.6.noarch
python3-3.6.4-1.2.x86_64
python3-Pygments-2.2.0-53.7.noarch
python3-appdirs-1.4.3-8.10.noarch
python3-base-3.6.4-1.3.x86_64
python3-devel-3.6.4-1.3.x86_64
python3-packaging-16.8-20.15.noarch
python3-pyparsing-2.2.0-1.15.noarch
python3-setuptools-38.4.0-112.2.noarch
python3-six-1.11.0-66.6.noarch
```

And here the tests failing with markdown2 2.3.5
**Summary**
```
+ echo python2
+ export PYTHONPATH=/home/abuild/rpmbuild/BUILDROOT/python-markdown2-2.3.5-0.x86_64/usr/lib/python2.7/site-packages
+ PYTHONPATH=/home/abuild/rpmbuild/BUILDROOT/python-markdown2-2.3.5-0.x86_64/usr/lib/python2.7/site-packages
+ python2 test.py -- -knownfailure
markdown2/tm/ampersands [htmlentities] ... ok
markdown2/tm/auto_link ... ok
markdown2/tm/auto_link_email_with_underscore [issue26] ... ok
markdown2/tm/auto_link_safe_mode [issue7, safe_mode] ... ok
markdown2/tm/basic_safe_mode [safe_mode] ... ok
markdown2/tm/basic_safe_mode_escape [safe_mode] ... ok
markdown2/tm/blockquote ... ok
markdown2/tm/blockquote_with_pre ... ok
markdown2/tm/break_on_newline ... ok
markdown2/tm/code_block_with_tabs [fromphpmarkdown] ... ok
markdown2/tm/code_blocks_leading_line ... ok
markdown2/tm/code_safe_emphasis [code_friendly] ... ok
markdown2/tm/codeblock ... ok
markdown2/tm/codespans ... ok
markdown2/tm/codespans_safe_mode [issue9, safe_mode] ... ok
markdown2/tm/cuddled_list_indented [smedberg, issue36, extra, cuddle-lists] ... ok
markdown2/tm/cuddled_para_and_list [smedberg, issue33, extra, cuddle-lists] ... ok
markdown2/tm/cuddled_with_para ... ok
markdown2/tm/demote_headers ... ok
markdown2/tm/emacs_head_vars [emacs] ... ok
markdown2/tm/emacs_tail_vars [emacs] ... ok
markdown2/tm/emphasis ... ok
markdown2/tm/escapes [issue15] ... ok
markdown2/tm/fenced_code_blocks_issue113 [extra, fenced-code-blocks, issue113] ... ok
markdown2/tm/fenced_code_blocks_issue127 [extra, fenced-code-blocks, issue127] ... ok
markdown2/tm/fenced_code_blocks_issue135 [extra, fenced-code-blocks, issue135] ... ok
markdown2/tm/fenced_code_blocks_issue161 [extra, fenced-code-blocks] ... ok
markdown2/tm/fenced_code_blocks_issue86 [extra, fenced-code-blocks, issue86, issue84, issue87] ... ok
markdown2/tm/fenced_code_blocks_leading_lang_space [extra, fenced-code-blocks] ... FAIL
markdown2/tm/fenced_code_blocks_leading_line [extra, fenced-code-blocks] ... ok
markdown2/tm/fenced_code_blocks_leading_paragraph [extra, fenced-code-blocks] ... ok
markdown2/tm/fenced_code_blocks_safe_highlight [extra, fenced-code-blocks, pygments] ... FAIL
markdown2/tm/fenced_code_blocks_simple [extra, fenced-code-blocks] ... ok
markdown2/tm/fenced_code_blocks_syntax_highlighting [extra, fenced-code-blocks, pygments] ... FAIL
markdown2/tm/fenced_code_blocks_syntax_indentation [extra, fenced-code-blocks, pygments, indentation] ... FAIL
markdown2/tm/footnotes ... ok
markdown2/tm/footnotes_custom ... ok
markdown2/tm/footnotes_letters [footnotes] ... ok
markdown2/tm/footnotes_markup [footnotes] ... ok
markdown2/tm/footnotes_safe_mode_escape [safe_mode, footnotes] ... ok
markdown2/tm/footnotes_underscores [issue27] ... ok
markdown2/tm/header ... ok
markdown2/tm/header_atx_no_preceeding_space ... ok
markdown2/tm/header_ids_1 [extra, header-ids] ... ok
markdown2/tm/header_ids_2 [extra, header-ids] ... ok
markdown2/tm/header_ids_3 [extra, header-ids] ... ok
markdown2/tm/header_ids_4 [extra, header-ids, unicode] ... ok
markdown2/tm/header_ids_5 [extra, header-ids] ... ok
markdown2/tm/headers_tag_friendly [extra, tag-friendly] ... ok
markdown2/tm/highlightjs_lang [extra, fenced-code-blocks, highlightjs-lang] ... ok
markdown2/tm/hr ... ok
markdown2/tm/hr_length ... ok
markdown2/tm/hr_spaces ... ok
markdown2/tm/html5_block_tags [issue57] ... ok
markdown2/tm/img_in_link ... ok
markdown2/tm/incorrect_list_parse [issue165] ... ok
markdown2/tm/inline_links ... ok
markdown2/tm/issue21_gt_escaping [issue21, escape] ... ok
markdown2/tm/issue2_safe_mode_borks_markup [safe_mode] ... ok
markdown2/tm/issue3_bad_code_color_hack [extra, code-color, unicode, pygments, issue3] ... FAIL
markdown2/tm/issue54_escape_link_title [issue54] ... ok
markdown2/tm/link_defn_alt_title_delims ... ok
markdown2/tm/link_defn_spaces_in_url [issue18] ... ok
markdown2/tm/link_nofollow [extras, issue74, issue104] ... ok
markdown2/tm/link_patterns ... ok
markdown2/tm/link_patterns_double_hit [link_patterns] ... ok
markdown2/tm/link_patterns_edge_cases [link_patterns] ... ok
markdown2/tm/link_with_blank [extras, issue213] ... ok
markdown2/tm/lists ... ok
markdown2/tm/lists2 [issue16] ... ok
markdown2/tm/long_link [issue24] ... ok
markdown2/tm/markdown_in_html [markdown-in-html, toc, extra] ... ok
markdown2/tm/metadata [extra, metadata, issue78] ... ok
markdown2/tm/metadata2 [extra, metadata, issue78] ... ok
markdown2/tm/mismatched_footnotes ... ok
markdown2/tm/missing_link_defn ... ok
markdown2/tm/nested_list ... ok
markdown2/tm/nested_list_safe_mode [issue9, safe_mode] ... ok
markdown2/tm/not_quite_a_list ... ok
markdown2/tm/numbering ... ok
markdown2/tm/parens_in_url_4 [fromphpmarkdown] ... ok
markdown2/tm/pi_and_xinclude [xml, pi, xinclude] ... ok
markdown2/tm/pyshell ... ok
markdown2/tm/raw_html ... ok
markdown2/tm/ref_links ... ok
markdown2/tm/smarty_pants [smarty-pants, issue42] ... ok
markdown2/tm/smarty_pants_image_links [smarty-pants, issue76] ... ok
markdown2/tm/spoiler ... ok
markdown2/tm/strike [extras, strike] ... ok
markdown2/tm/sublist-ordered-para [issue67] ... ok
markdown2/tm/sublist-para [questionable] ... ok
markdown2/tm/syntax_color [extra, code-color, pygments] ... FAIL
markdown2/tm/syntax_color_opts [extra, code-color, pygments] ... FAIL
markdown2/tm/tables [extra, tables] ... ok
markdown2/tm/task_list [extra, task_list, issue216] ... ok
markdown2/tm/toc_1 [toc, extra] ... ok
markdown2/tm/toc_2 [toc, extra] ... ok
markdown2/tm/toc_3 [toc, extra] ... ok
markdown2/tm/toc_4 [toc, extra] ... ok
markdown2/tm/tricky_anchors ... ok
markdown2/tm/two_comments ... ok
markdown2/tm/underline_in_autolink ... ok
markdown2/tm/wiki_tables [extra, wiki-tables, issue66] ... ok
markdown2/tm/xss_quotes [issue30] ... ok
markdown2/direct/code_in_strong [code, strong] ... ok
markdown2/direct/pre ... ok
markdown2/direct/russian [unicode, issue3] ... ok
markdown2/direct/slow_hr [perf] ... ok
markdown2/direct/starter_pre [pre, recipes] ... ok
markdown2/doctests/api ... ok
markdown2/doctests/internal ... ok
```
**Details**
```
markdown2/doctests/api ... ok
markdown2/doctests/internal ... ok

======================================================================
FAIL: markdown2/tm/fenced_code_blocks_leading_lang_space [extra, fenced-code-blocks]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    ```.python\n
    if.True:\n
    ....print."hi"\n
    ```\n
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><span></span><code><span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s2">&quot;hi&quot;</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><code><span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s">&quot;hi&quot;</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/fenced_code_blocks_leading_lang_space.html
    +++ markdown2 ./tm-cases/fenced_code_blocks_leading_lang_space.text
    @@ -1,3 +1,3 @@
    -<div class="codehilite"><pre><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    -    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    +    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
     </code></pre></div>


======================================================================
FAIL: markdown2/tm/fenced_code_blocks_safe_highlight [extra, fenced-code-blocks, pygments]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    ```python\n
    if.True:\n
    ....print."hi"\n
    ```\n
    \n
    That's.using.the.*fenced-code-blocks*.extra.with.Python\n
    syntax.coloring,.if.`pygments`.is.installed..See\n
    <http://github.github.com/github-flavored-markdown/>.\n
    \n
    ```ruby\n
    def.foo\n
    ....puts."hi"\n
    end\n
    ```\n
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><span></span><code><span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s2">&quot;hi&quot;</span>\n
    </code></pre></div>\n
    \n
    <p>That's.using.the.<em>fenced-code-blocks</em>.extra.with.Python\n
    syntax.coloring,.if.<code>pygments</code>.is.installed..See\n
    <a.href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>\n
    \n
    <div.class="codehilite"><pre><span></span><code><span.class="k">def</span>.<span.class="nf">foo</span>\n
    ....<span.class="nb">puts</span>.<span.class="s2">&quot;hi&quot;</span>\n
    <span.class="k">end</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><code><span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s">&quot;hi&quot;</span>\n
    </code></pre></div>\n
    \n
    <p>That's.using.the.<em>fenced-code-blocks</em>.extra.with.Python\n
    syntax.coloring,.if.<code>pygments</code>.is.installed..See\n
    <a.href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>\n
    \n
    <div.class="codehilite"><pre><code><span.class="k">def</span>.<span.class="nf">foo</span>\n
    ....<span.class="nb">puts</span>.<span.class="s2">&quot;hi&quot;</span>\n
    <span.class="k">end</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/fenced_code_blocks_safe_highlight.html
    +++ markdown2 ./tm-cases/fenced_code_blocks_safe_highlight.text
    @@ -1,12 +1,12 @@
    -<div class="codehilite"><pre><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    -    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    +    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
     </code></pre></div>
     
     <p>That's using the <em>fenced-code-blocks</em> extra with Python
     syntax coloring, if <code>pygments</code> is installed. See
     <a href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>
     
    -<div class="codehilite"><pre><code><span class="k">def</span> <span class="nf">foo</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">def</span> <span class="nf">foo</span>
         <span class="nb">puts</span> <span class="s2">&quot;hi&quot;</span>
     <span class="k">end</span>
     </code></pre></div>


======================================================================
FAIL: markdown2/tm/fenced_code_blocks_syntax_highlighting [extra, fenced-code-blocks, pygments]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    ```python\n
    if.True:\n
    ....print."hi"\n
    ```\n
    \n
    That's.using.the.*fenced-code-blocks*.extra.with.Python\n
    syntax.coloring,.if.`pygments`.is.installed..See\n
    <http://github.github.com/github-flavored-markdown/>.\n
    \n
    ```ruby\n
    def.foo\n
    ....puts."hi"\n
    end\n
    ```\n
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><span></span><code><span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s2">&quot;hi&quot;</span>\n
    </code></pre></div>\n
    \n
    <p>That's.using.the.<em>fenced-code-blocks</em>.extra.with.Python\n
    syntax.coloring,.if.<code>pygments</code>.is.installed..See\n
    <a.href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>\n
    \n
    <div.class="codehilite"><pre><span></span><code><span.class="k">def</span>.<span.class="nf">foo</span>\n
    ....<span.class="nb">puts</span>.<span.class="s2">&quot;hi&quot;</span>\n
    <span.class="k">end</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><code><span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s">&quot;hi&quot;</span>\n
    </code></pre></div>\n
    \n
    <p>That's.using.the.<em>fenced-code-blocks</em>.extra.with.Python\n
    syntax.coloring,.if.<code>pygments</code>.is.installed..See\n
    <a.href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>\n
    \n
    <div.class="codehilite"><pre><code><span.class="k">def</span>.<span.class="nf">foo</span>\n
    ....<span.class="nb">puts</span>.<span.class="s2">&quot;hi&quot;</span>\n
    <span.class="k">end</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/fenced_code_blocks_syntax_highlighting.html
    +++ markdown2 ./tm-cases/fenced_code_blocks_syntax_highlighting.text
    @@ -1,12 +1,12 @@
    -<div class="codehilite"><pre><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    -    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    +    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
     </code></pre></div>
     
     <p>That's using the <em>fenced-code-blocks</em> extra with Python
     syntax coloring, if <code>pygments</code> is installed. See
     <a href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>
     
    -<div class="codehilite"><pre><code><span class="k">def</span> <span class="nf">foo</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">def</span> <span class="nf">foo</span>
         <span class="nb">puts</span> <span class="s2">&quot;hi&quot;</span>
     <span class="k">end</span>
     </code></pre></div>


======================================================================
FAIL: markdown2/tm/fenced_code_blocks_syntax_indentation [extra, fenced-code-blocks, pygments, indentation]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    ```python\n
    def.foo():\n
    ....print."foo"\n
    \n
    ....print."bar"\n
    ```\n
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><span></span><code><span.class="k">def</span>.<span.class="nf">foo</span><span.class="p">():</span>\n
    ....<span.class="k">print</span>.<span.class="s2">&quot;foo&quot;</span>\n
    \n
    ....<span.class="k">print</span>.<span.class="s2">&quot;bar&quot;</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <div.class="codehilite"><pre><code><span.class="k">def</span>.<span.class="nf">foo</span><span.class="p">():</span>\n
    ....<span.class="k">print</span>.<span.class="s">&quot;foo&quot;</span>\n
    \n
    ....<span.class="k">print</span>.<span.class="s">&quot;bar&quot;</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/fenced_code_blocks_syntax_indentation.html
    +++ markdown2 ./tm-cases/fenced_code_blocks_syntax_indentation.text
    @@ -1,5 +1,5 @@
    -<div class="codehilite"><pre><code><span class="k">def</span> <span class="nf">foo</span><span class="p">():</span>
    -    <span class="k">print</span> <span class="s">&quot;foo&quot;</span>
    +<div class="codehilite"><pre><span></span><code><span class="k">def</span> <span class="nf">foo</span><span class="p">():</span>
    +    <span class="k">print</span> <span class="s2">&quot;foo&quot;</span>
     
    -    <span class="k">print</span> <span class="s">&quot;bar&quot;</span>
    +    <span class="k">print</span> <span class="s2">&quot;bar&quot;</span>
     </code></pre></div>


======================================================================
FAIL: markdown2/tm/issue3_bad_code_color_hack [extra, code-color, unicode, pygments, issue3]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    <!--.-*-.coding:.utf-8.-*-.-->\n
    \n
    ##.\u0437\u0430\u0433\u043e\u043b\u043e\u0432\u043e\u043a\n
    \n
    Example.from.<http://code.google.com/p/python-markdown2/issues/detail?id=3#c8>.\n
    \n
    Some.python.code:\n
    \n
    ....:::python\n
    ....#.\u043a\u043e\u043c\u043c\u0435\u043d\u0442\u0430\u0440\u0438\u0439\n
    ....if.True:\n
    ........print."hi"
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <!--.-*-.coding:.utf-8.-*-.-->\n
    \n
    <h2>\u0437\u0430\u0433\u043e\u043b\u043e\u0432\u043e\u043a</h2>\n
    \n
    <p>Example.from.<a.href="http://code.google.com/p/python-markdown2/issues/detail?id=3#c8">http://code.google.com/p/python-markdown2/issues/detail?id=3#c8</a>.</p>\n
    \n
    <p>Some.python.code:</p>\n
    \n
    <div.class="codehilite"><pre><span></span><code><span.class="c1">#.\u043a\u043e\u043c\u043c\u0435\u043d\u0442\u0430\u0440\u0438\u0439</span>\n
    <span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s2">&quot;hi&quot;</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <!--.-*-.coding:.utf-8.-*-.-->\n
    \n
    <h2>\u0437\u0430\u0433\u043e\u043b\u043e\u0432\u043e\u043a</h2>\n
    \n
    <p>Example.from.<a.href="http://code.google.com/p/python-markdown2/issues/detail?id=3#c8">http://code.google.com/p/python-markdown2/issues/detail?id=3#c8</a>.</p>\n
    \n
    <p>Some.python.code:</p>\n
    \n
    <div.class="codehilite"><pre><code><span.class="c">#.\u043a\u043e\u043c\u043c\u0435\u043d\u0442\u0430\u0440\u0438\u0439</span>\n
    <span.class="k">if</span>.<span.class="bp">True</span><span.class="p">:</span>\n
    ....<span.class="k">print</span>.<span.class="s">&quot;hi&quot;</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/issue3_bad_code_color_hack.html
    +++ markdown2 ./tm-cases/issue3_bad_code_color_hack.text
    @@ -6,7 +6,7 @@
     
     <p>Some python code:</p>
     
    -<div class="codehilite"><pre><code><span class="c"># \u043a\u043e\u043c\u043c\u0435\u043d\u0442\u0430\u0440\u0438\u0439</span>
    +<div class="codehilite"><pre><span></span><code><span class="c1"># \u043a\u043e\u043c\u043c\u0435\u043d\u0442\u0430\u0440\u0438\u0439</span>
     <span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
    -    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
    +    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
     </code></pre></div>


======================================================================
FAIL: markdown2/tm/syntax_color [extra, code-color, pygments]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    Here.is.some.sample.code:\n
    \n
    ....:::python\n
    ....import.sys\n
    ....def.main(argv=sys.argv):\n
    ........logging.basicConfig()\n
    ........log.info('hi')\n
    \n
    and:\n
    \n
    ....:::ruby\n
    ....use.'zlib'\n
    ....sub.main(argv)\n
    ........puts.'hi'\n
    ....end\n
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <p>Here.is.some.sample.code:</p>\n
    \n
    <div.class="codehilite"><pre><span></span><code><span.class="kn">import</span>.<span.class="nn">sys</span>\n
    <span.class="k">def</span>.<span.class="nf">main</span><span.class="p">(</span><span.class="n">argv</span><span.class="o">=</span><span.class="n">sys</span><span.class="o">.</span><span.class="n">argv</span><span.class="p">):</span>\n
    ....<span.class="n">logging</span><span.class="o">.</span><span.class="n">basicConfig</span><span.class="p">()</span>\n
    ....<span.class="n">log</span><span.class="o">.</span><span.class="n">info</span><span.class="p">(</span><span.class="s1">&#39;hi&#39;</span><span.class="p">)</span>\n
    </code></pre></div>\n
    \n
    <p>and:</p>\n
    \n
    <div.class="codehilite"><pre><span></span><code><span.class="n">use</span>.<span.class="s1">&#39;zlib&#39;</span>\n
    <span.class="nb">sub</span>.<span.class="n">main</span><span.class="p">(</span><span.class="n">argv</span><span.class="p">)</span>\n
    ....<span.class="nb">puts</span>.<span.class="s1">&#39;hi&#39;</span>\n
    <span.class="k">end</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <p>Here.is.some.sample.code:</p>\n
    \n
    <div.class="codehilite"><pre><code><span.class="kn">import</span>.<span.class="nn">sys</span>\n
    <span.class="k">def</span>.<span.class="nf">main</span><span.class="p">(</span><span.class="n">argv</span><span.class="o">=</span><span.class="n">sys</span><span.class="o">.</span><span.class="n">argv</span><span.class="p">):</span>\n
    ....<span.class="n">logging</span><span.class="o">.</span><span.class="n">basicConfig</span><span.class="p">()</span>\n
    ....<span.class="n">log</span><span.class="o">.</span><span.class="n">info</span><span.class="p">(</span><span.class="s">&#39;hi&#39;</span><span.class="p">)</span>\n
    </code></pre></div>\n
    \n
    <p>and:</p>\n
    \n
    <div.class="codehilite"><pre><code><span.class="n">use</span>.<span.class="s1">&#39;zlib&#39;</span>\n
    <span.class="nb">sub</span>.<span.class="n">main</span><span.class="p">(</span><span.class="n">argv</span><span.class="p">)</span>\n
    ....<span.class="nb">puts</span>.<span.class="s1">&#39;hi&#39;</span>\n
    <span.class="k">end</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/syntax_color.html
    +++ markdown2 ./tm-cases/syntax_color.text
    @@ -1,14 +1,14 @@
     <p>Here is some sample code:</p>
     
    -<div class="codehilite"><pre><code><span class="kn">import</span> <span class="nn">sys</span>
    +<div class="codehilite"><pre><span></span><code><span class="kn">import</span> <span class="nn">sys</span>
     <span class="k">def</span> <span class="nf">main</span><span class="p">(</span><span class="n">argv</span><span class="o">=</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">):</span>
         <span class="n">logging</span><span class="o">.</span><span class="n">basicConfig</span><span class="p">()</span>
    -    <span class="n">log</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s">&#39;hi&#39;</span><span class="p">)</span>
    +    <span class="n">log</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s1">&#39;hi&#39;</span><span class="p">)</span>
     </code></pre></div>
     
     <p>and:</p>
     
    -<div class="codehilite"><pre><code><span class="n">use</span> <span class="s1">&#39;zlib&#39;</span>
    +<div class="codehilite"><pre><span></span><code><span class="n">use</span> <span class="s1">&#39;zlib&#39;</span>
     <span class="nb">sub</span> <span class="n">main</span><span class="p">(</span><span class="n">argv</span><span class="p">)</span>
         <span class="nb">puts</span> <span class="s1">&#39;hi&#39;</span>
     <span class="k">end</span>


======================================================================
FAIL: markdown2/tm/syntax_color_opts [extra, code-color, pygments]
----------------------------------------------------------------------
Traceback (most recent call last):
  File "./test_markdown2.py", line 202, in <lambda>
    metadata_path=m)
  File "./test_markdown2.py", line 93, in _assertMarkdownPath
    self._assertMarkdown(text, html, text_path, html_path, opts=opts, **extra)
  File "./test_markdown2.py", line 143, in _assertMarkdown
    self.assertEqual(python_norm_html, norm_html, errmsg)
AssertionError: markdown2.py didn't produce the expected HTML:
  ---- text (escaping: .=space, \n=newline) ----
    Here.is.some.sample.code:\n
    \n
    ....:::python\n
    ....import.sys\n
    ....def.main(argv=sys.argv):\n
    ........logging.basicConfig()\n
    ........log.info('hi')\n
    \n
    and:\n
    \n
    ....:::ruby\n
    ....use.'zlib'\n
    ....sub.main(argv)\n
    ........puts.'hi'\n
    ....end\n
  ---- Python markdown2.py HTML (escaping: .=space, \n=newline) ----
    <p>Here.is.some.sample.code:</p>\n
    \n
    <div.class="codehilite".style="background:.#f8f8f8"><pre.style="line-height:.125%"><span></span><code><span.style="color:.#008000;.font-weight:.bold">import</span>.<span.style="color:.#0000FF;.font-weight:.bold">sys</span>\n
    <span.style="color:.#008000;.font-weight:.bold">def</span>.<span.style="color:.#0000FF">main</span>(argv<span.style="color:.#666666">=</span>sys<span.style="color:.#666666">.</span>argv):\n
    ....logging<span.style="color:.#666666">.</span>basicConfig()\n
    ....log<span.style="color:.#666666">.</span>info(<span.style="color:.#BA2121">&#39;hi&#39;</span>)\n
    </code></pre></div>\n
    \n
    <p>and:</p>\n
    \n
    <div.class="codehilite".style="background:.#f8f8f8"><pre.style="line-height:.125%"><span></span><code>use.<span.style="color:.#BA2121">&#39;zlib&#39;</span>\n
    <span.style="color:.#008000">sub</span>.main(argv)\n
    ....<span.style="color:.#008000">puts</span>.<span.style="color:.#BA2121">&#39;hi&#39;</span>\n
    <span.style="color:.#008000;.font-weight:.bold">end</span>\n
    </code></pre></div>\n
  ---- expected HTML (escaping: .=space, \n=newline) ----
    <p>Here.is.some.sample.code:</p>\n
    \n
    <div.class="codehilite".style="background:.#f8f8f8"><pre.style="line-height:.125%"><code><span.style="color:.#008000;.font-weight:.bold">import</span>.<span.style="color:.#0000FF;.font-weight:.bold">sys</span>\n
    <span.style="color:.#008000;.font-weight:.bold">def</span>.<span.style="color:.#0000FF">main</span>(argv<span.style="color:.#666666">=</span>sys<span.style="color:.#666666">.</span>argv):\n
    ....logging<span.style="color:.#666666">.</span>basicConfig()\n
    ....log<span.style="color:.#666666">.</span>info(<span.style="color:.#BA2121">&#39;hi&#39;</span>)\n
    </code></pre></div>\n
    \n
    <p>and:</p>\n
    \n
    <div.class="codehilite".style="background:.#f8f8f8"><pre.style="line-height:.125%"><code>use.<span.style="color:.#BA2121">&#39;zlib&#39;</span>\n
    <span.style="color:.#008000">sub</span>.main(argv)\n
    ....<span.style="color:.#008000">puts</span>.<span.style="color:.#BA2121">&#39;hi&#39;</span>\n
    <span.style="color:.#008000;.font-weight:.bold">end</span>\n
    </code></pre></div>\n
  ---- diff ----
    --- ./tm-cases/syntax_color_opts.html
    +++ markdown2 ./tm-cases/syntax_color_opts.text
    @@ -1,6 +1,6 @@
     <p>Here is some sample code:</p>

    -<div class="codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><code><span style="color: #008000; font-weight: bold">import</span> <span style="color: #0000FF; font-weight: bold">sys</span>
    +<div class="codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><span></span><code><span style="color: #008000; font-weight: bold">import</span> <span style="color: #0000FF; font-weight: bold">sys</span>
     <span style="color: #008000; font-weight: bold">def</span> <span style="color: #0000FF">main</span>(argv<span style="color: #666666">=</span>sys<span style="color: #666666">.</span>argv):
         logging<span style="color: #666666">.</span>basicConfig()
         log<span style="color: #666666">.</span>info(<span style="color: #BA2121">&#39;hi&#39;</span>)
    @@ -8,7 +8,7 @@

     <p>and:</p>

    -<div class="codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><code>use <span style="color: #BA2121">&#39;zlib&#39;</span>
    +<div class="codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><span></span><code>use <span style="color: #BA2121">&#39;zlib&#39;</span>
     <span style="color: #008000">sub</span> main(argv)
         <span style="color: #008000">puts</span> <span style="color: #BA2121">&#39;hi&#39;</span>
     <span style="color: #008000; font-weight: bold">end</span>


----------------------------------------------------------------------
Ran 111 tests in 0.157s

FAILED (7 failures)
error: Bad exit status from /var/tmp/rpm-tmp.vaPyri (%check)
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -8,6 +8,6 @@ python:
   - "3.5"
   - "3.6"
 # command to install dependencies
-install: pip install Pygments==1.6
+install: pip install Pygments>=2.1.1
 # command to run tests
 script: make testone

--- a/test/tm-cases/fenced_code_blocks_leading_lang_space.html
+++ b/test/tm-cases/fenced_code_blocks_leading_lang_space.html
@@ -1,3 +1,3 @@
-<div class="codehilite"><pre><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
-    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
+<div class="codehilite"><pre><span></span><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
+    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
 </code></pre></div>

--- a/test/tm-cases/fenced_code_blocks_safe_highlight.html
+++ b/test/tm-cases/fenced_code_blocks_safe_highlight.html
@@ -1,12 +1,12 @@
-<div class="codehilite"><pre><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
-    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
+<div class="codehilite"><pre><span></span><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
+    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
 </code></pre></div>
 
 <p>That's using the <em>fenced-code-blocks</em> extra with Python
 syntax coloring, if <code>pygments</code> is installed. See
 <a href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>
 
-<div class="codehilite"><pre><code><span class="k">def</span> <span class="nf">foo</span>
+<div class="codehilite"><pre><span></span><code><span class="k">def</span> <span class="nf">foo</span>
     <span class="nb">puts</span> <span class="s2">&quot;hi&quot;</span>
 <span class="k">end</span>
 </code></pre></div>

--- a/test/tm-cases/fenced_code_blocks_syntax_highlighting.html
+++ b/test/tm-cases/fenced_code_blocks_syntax_highlighting.html
@@ -1,12 +1,12 @@
-<div class="codehilite"><pre><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
-    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
+<div class="codehilite"><pre><span></span><code><span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
+    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
 </code></pre></div>
 
 <p>That's using the <em>fenced-code-blocks</em> extra with Python
 syntax coloring, if <code>pygments</code> is installed. See
 <a href="http://github.github.com/github-flavored-markdown/">http://github.github.com/github-flavored-markdown/</a>.</p>
 
-<div class="codehilite"><pre><code><span class="k">def</span> <span class="nf">foo</span>
+<div class="codehilite"><pre><span></span><code><span class="k">def</span> <span class="nf">foo</span>
     <span class="nb">puts</span> <span class="s2">&quot;hi&quot;</span>
 <span class="k">end</span>
 </code></pre></div>

--- a/test/tm-cases/fenced_code_blocks_syntax_indentation.html
+++ b/test/tm-cases/fenced_code_blocks_syntax_indentation.html
@@ -1,5 +1,5 @@
-<div class="codehilite"><pre><code><span class="k">def</span> <span class="nf">foo</span><span class="p">():</span>
-    <span class="k">print</span> <span class="s">&quot;foo&quot;</span>
+<div class="codehilite"><pre><span></span><code><span class="k">def</span> <span class="nf">foo</span><span class="p">():</span>
+    <span class="k">print</span> <span class="s2">&quot;foo&quot;</span>
 
-    <span class="k">print</span> <span class="s">&quot;bar&quot;</span>
+    <span class="k">print</span> <span class="s2">&quot;bar&quot;</span>
 </code></pre></div>

--- a/test/tm-cases/issue3_bad_code_color_hack.html
+++ b/test/tm-cases/issue3_bad_code_color_hack.html
@@ -6,7 +6,7 @@
 
 <p>Some python code:</p>
 
-<div class="codehilite"><pre><code><span class="c"># комментарий</span>
+<div class="codehilite"><pre><span></span><code><span class="c1"># комментарий</span>
 <span class="k">if</span> <span class="bp">True</span><span class="p">:</span>
-    <span class="k">print</span> <span class="s">&quot;hi&quot;</span>
+    <span class="k">print</span> <span class="s2">&quot;hi&quot;</span>
 </code></pre></div>

--- a/test/tm-cases/syntax_color.html
+++ b/test/tm-cases/syntax_color.html
@@ -1,14 +1,14 @@
 <p>Here is some sample code:</p>
 
-<div class="codehilite"><pre><code><span class="kn">import</span> <span class="nn">sys</span>
+<div class="codehilite"><pre><span></span><code><span class="kn">import</span> <span class="nn">sys</span>
 <span class="k">def</span> <span class="nf">main</span><span class="p">(</span><span class="n">argv</span><span class="o">=</span><span class="n">sys</span><span class="o">.</span><span class="n">argv</span><span class="p">):</span>
     <span class="n">logging</span><span class="o">.</span><span class="n">basicConfig</span><span class="p">()</span>
-    <span class="n">log</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s">&#39;hi&#39;</span><span class="p">)</span>
+    <span class="n">log</span><span class="o">.</span><span class="n">info</span><span class="p">(</span><span class="s1">&#39;hi&#39;</span><span class="p">)</span>
 </code></pre></div>
 
 <p>and:</p>
 
-<div class="codehilite"><pre><code><span class="n">use</span> <span class="s1">&#39;zlib&#39;</span>
+<div class="codehilite"><pre><span></span><code><span class="n">use</span> <span class="s1">&#39;zlib&#39;</span>
 <span class="nb">sub</span> <span class="n">main</span><span class="p">(</span><span class="n">argv</span><span class="p">)</span>
     <span class="nb">puts</span> <span class="s1">&#39;hi&#39;</span>
 <span class="k">end</span>

--- a/test/tm-cases/syntax_color_opts.html
+++ b/test/tm-cases/syntax_color_opts.html
@@ -1,6 +1,6 @@
 <p>Here is some sample code:</p>
 
-<div class="codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><code><span style="color: #008000; font-weight: bold">import</span> <span style="color: #0000FF; font-weight: bold">sys</span>
+<div class="codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><span></span><code><span style="color: #008000; font-weight: bold">import</span> <span style="color: #0000FF; font-weight: bold">sys</span>
 <span style="color: #008000; font-weight: bold">def</span> <span style="color: #0000FF">main</span>(argv<span style="color: #666666">=</span>sys<span style="color: #666666">.</span>argv):
     logging<span style="color: #666666">.</span>basicConfig()
     log<span style="color: #666666">.</span>info(<span style="color: #BA2121">&#39;hi&#39;</span>)
@@ -8,7 +8,7 @@
 
 <p>and:</p>
 
-<div class="codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><code>use <span style="color: #BA2121">&#39;zlib&#39;</span>
+<div class="codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><span></span><code>use <span style="color: #BA2121">&#39;zlib&#39;</span>
 <span style="color: #008000">sub</span> main(argv)
     <span style="color: #008000">puts</span> <span style="color: #BA2121">&#39;hi&#39;</span>
 <span style="color: #008000; font-weight: bold">end</span>

